### PR TITLE
okx: updating createOrder and isPostOnly

### PR DESCRIPTION
**Pull request by DoctorSlimm**

### CODE REVIEW

File: js/okx.js [hunk 1]
```diff
@@ -1927,79 +1927,71 @@ module.exports = class okx extends Exchange {
...
         };
-        const tdMode = this.safeStringLower (params, 'tdMode');
-        if (market['spot']) {
+        const spot = market['spot'];
+        const swap = market['swap'];
+        const future = market['future'];
+        const contract = market['contract'];
+        const stopPrice = this.safeString2 (params, 'stopPrice', 'triggerPx');
+        const timeInForce = this.safeString (params, 'timeInForce', 'GTC');
+        const takeProfitPrice = this.safeString2 (params, 'takeProfitPrice', 'tpTriggerPrice');
```
Comment (kroitor) [inline js/okx.js:1960]: Typo fix:
```suggestion
        const takeProfitPrice = this.safeString2 (params, 'takeProfitPrice', 'tpTriggerPx');
```

File: js/okx.js [hunk 2]
```diff
@@ -2028,63 +2020,85 @@ module.exports = class okx extends Exchange {
...
                 price = -1;
             }
             request['orderPx'] = this.priceToPrecision (symbol, price);
+        } else if (conditional) {
+            method = 'privatePostTradeOrderAlgo';
+            request['ordType'] = 'conditional';
+            const twoWayCondition = ((takeProfitPrice !== undefined) && (stopLossPrice !== undefined));
+            if (type !== 'market') {
+                if ((twoWayCondition) && ((!slOrdPx) || (!tpOrdPx))) {
+                    throw new InvalidOrder (this.id + ' createOrder() will cannot use the same price for two-way conditional orders to be created, please supply tpOrdPx and slOrdPx params');
```
Comment (kroitor) [inline js/okx.js:2057]: Typo fix:
```suggestion
                    throw new InvalidOrder (this.id + ' createOrder() cannot use the same price for two-way conditional orders to be created, please supply takeProfitPrice and stopLossPrice or tpOrdPx and slOrdPx params');
```

**Comment by DoctorSlimm** [comment]
@kroitor okx createOrder conditional error message typo fix - they are not the same thing, the stopLossPrice and takeProfitPrice are the triggers and the slOrdPx and tpOrdPx are the limit order prices in the case where they're limit orders

**Comment by kroitor** [comment]
> @kroitor okx createOrder conditional error message typo fix - they are not the same thing, the stopLossPrice and takeProfitPrice are the triggers and the slOrdPx and tpOrdPx are the limit order prices in the case where they're limit orders - they have to be both supplied unless it's a market (both ways) order

Tbh, I'm not sure I fully understand. Assuming the terms from here: https://github.com/ccxt/ccxt/issues/12044#issuecomment-1067582000 – we should use `stopPrice` and `triggerPx` for trigger orders, and we should use `stopLossPrice` and `takeProfitPrice` (`slOrdPx` and `tpOrdPx`) for conditional closing orders, right?

**Comment by DoctorSlimm** [comment]
@kroitor yeah that is correct and slOrdPx and tpOrdPx are the order prices when the asset passes the SL and TP thresholds (and -1 if to be executed at market price)

**Comment by kroitor** [comment]
> @kroitor yeah that is correct and slOrdPx and tpOrdPx are the limit order prices (and -1 if to be executed at market price) for the orders which are placed on the book when the asset passes the SL and TP thresholds

So my edits were correct or not exactly ?

**Comment by DoctorSlimm** [comment]
@kroitor only changes to the InvalidOrder error throwing message are wrong in the commit titled "okx createOrder conditional error message typo fix"

**Pull request merged**